src/context/mod.rs

```diff
@@ -0,0 +1,152 @@
+use ckb_vm_definitions::instructions::Instruction;
+
+use super::Error;
+use crate::machine::SupportMachine;
+
+pub trait ExecutionContext<Mac: SupportMachine> {
+    fn initialize(&mut self, machine: &mut Mac) -> Result<(), Error> {
+        // We don't want to change param name to start with _ or others
```

Comment:
Can you shed some lights on the reasons?

Reply:
It's for IDE completion of trait impls. I don't really recall whether IDEs remove starting underscores of args when completing trait impl methods.

It also looks better in generated docs.

Reply:
Are there other projects that also leverage the same techniques? I wasn't sure if we should aid what the IDE should do well here.

Reply:
Not sure. But why not?

Reply:
https://internals.rust-lang.org/t/unused-parameters-in-default-trait-impls/10984

Reply:
I feel like we are self-inventing a weird pattern for a problem that 1) does not exist, or; 2) conventionally gets resolved in a different way.

I've seen the usage of `#[allow(unused_variables)]`, or the use of `let _ = machine;` before. But combining a clippy rule with drop here is not something people used earlier. To avoid future confusion, I would recommend switch to either of the above 2 solutions, also leave a more complete reason in comments as why we don't want to change param name to start with _

Reply:
It's not a big issue, but definitely exists. Drop and `let _ =` is the same but drop is more future proof than `let _` because clippy already have lints for drop but don't have lints for `let _` now. If clippy adds some lint for `let _ = drop_or_ref` in the future, we would need to add a clippy allow too.

Changing to `#[allow(unused_variables)]` might work, but it'll make the signature a lot noisier.

Comment can be added.

Reply:
Comment added.

Reply:
After a bit of research, I learned that [`let _ =` is not the same as `drop`](https://news.ycombinator.com/item?id=22449960) and is a bit more idiomatic for ignoring stuff. So it's unlikely that clippy will add lint for these, so I've changed this to use `let _ =`.

It can also be `_ = machine` but that's a relatively new language feature and will probably look foreign to many.

There's also concern about [accidentally ignoring other stuff](https://users.rust-lang.org/t/what-is-the-best-way-to-ignore-a-result/55187) when you only want to ignore a result. It's not really relevant here but is something that should be kept in mind when ignoring results. Personally I sometimes define a `ignore_error<E>(err: Result<(), E>)` function specifically for ignoring errors.